### PR TITLE
refactor: adapt Rsbuild environment config

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

bump rsbuild 1.0.0-alpha and use Rsbuild environment config
-  adapt Rsbuild environment config
-  compat serviceWorker environment in uni-builder
-  compat dev.startUrl / dev.beforeStartUrl / distPath.server / distPath.worker in uni-builder
-  support html minify and compat tools.htmlPlugin minify option
-  remove entry and target Options in createUniBuilder

## Related Links

https://github.com/web-infra-dev/rsbuild/discussions/2508

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/uni-builder/src/shared/plugins/htmlMinify.ts
Comment (chenjiahan) [inline packages/cli/uni-builder/src/shared/plugins/htmlMinify.ts:1]: What's the difference between this plugin and https://github.com/rspack-contrib/rsbuild-plugin-html-minifier-terser

File: packages/cli/uni-builder/tests/__snapshots__/styledComponents.test.ts.snap [hunk 1]
```diff
@@ -176,7 +178,7 @@ exports[`plugins/styled-components > should enable ssr when target contain node
               "<WORKSPACE>/node_modules/<PNPM_INNER>/babel-plugin-import/lib/index.js",
               {
                 "camel2DashComponentName": false,
-                "libraryDirectory": "lib",
+                "libraryDirectory": "es",
```
Comment (chenjiahan) [inline packages/cli/uni-builder/tests/__snapshots__/styledComponents.test.ts.snap:181]: should this be `lib` when target is node?

File: packages/cli/uni-builder/src/shared/plugins/htmlMinify.ts
Comment (9aoy) [reply packages/cli/uni-builder/src/shared/plugins/htmlMinify.ts:1]: Support applying `tools.htmlPlugin` minify option, and `tools.htmlPlugin` supports updating minifyConfig based on entryName, which is difficult to support in `rsbuild-plugin-html-minifier-terser`.

File: packages/cli/uni-builder/tests/__snapshots__/styledComponents.test.ts.snap (hunk 1 above)
Comment (9aoy) [reply packages/cli/uni-builder/tests/__snapshots__/styledComponents.test.ts.snap:181]: This is the snapshot of target web. Previously, when the rsbuild target had node, all library directories used lib, but it seems to me that only node should use lib and web should use es.

image

**Pull request merged**